### PR TITLE
KNL-1435 Improve force.url.secure implementation

### PR DESCRIPTION
**Pull request by master-bob**
Improves the use of the force.url.secure property to reduce the chances
of an exception.

### CODE REVIEW

File: kernel/api/src/main/java/org/sakaiproject/util/RequestFilter.java [hunk 1]
```diff
@@ -278,7 +278,7 @@ public static String serverUrl(HttpServletRequest req)
 
 		// if force.url.secure is set (to a https port number), use https and this port
 		String forceSecure = System.getProperty("sakai.force.url.secure");
-		if (forceSecure != null && !"".equals(forceSecure)) {
+		if (forceSecure != null && forceSecure.isEmpty() && !"0".equals(forceSecure)) {
```
Comment (ottenhoff) [inline kernel/api/src/main/java/org/sakaiproject/util/RequestFilter.java]: aren't you looking for !forceSecure.isEmpty()  ?

Comment (master-bob) [inline kernel/api/src/main/java/org/sakaiproject/util/RequestFilter.java]: Yes! :blush:

Comment (master-bob) [inline kernel/api/src/main/java/org/sakaiproject/util/RequestFilter.java]: Fixed in latest commit (force pushed).

**Comment by bjones86** [comment]
LGTM

**Comment by jonespm** [comment]
I'm kind of late to the review, but I think if you used
int forceSecureInt = NumberUtils.toInt(forceSecure);
if (forceSecureInt > 0 && forceSecureInt <= 65535) {
. . . //number is okay
}

It would check for a max bound too and be a little cleaner? NumberUtils is nice because it just returns a zero if the conversion fails for anything, which is fine in this case.

**Comment by master-bob** [comment]
That seems like a much better solution! I can make that change Monday.

Leonardo C.

**Comment by master-bob** [comment]
@jonespm , doing so will require the addition of the 'commons-math3' library to the kernel. Is this acceptable?

**Comment by ottenhoff** [comment]
Not worth it if this would require moving a lib in .....

**Comment by jonespm** [comment]
I updated samigo to commons-math3 and that's the only other thing using it (nothing else is using commons-math). I feel like if more things used it and we just put commons-math3 into shared for 11 it would be fine. There's a lot of useful things in there that could simplify some other code.

**Comment by master-bob** [comment]
@jonespm, I definitely see your point and I agree. But I think that might be out of scope for this ticket ... ?

**Comment by jonespm** [comment]
I'd personally vote for making it a dependency in kernel here, then I'll move it into shared deploy in a new jira right afterward. But if @ottenhoff  wants to keep it out I'm okay with that too.

**Comment by ottenhoff** [comment]
I don't have a strong opinion here.... as long as we can get it done today.....

**Comment by jonespm** [comment]
@master-bob Just go for it, put the dependency locally in kernel-api and I'll move it to shared later today.

        <dependency>
            <groupId>org.apache.commons</groupId>
            <artifactId>commons-math3</artifactId>
        </dependency

**Comment by master-bob** [comment]
I made a mistake. The library needed is 'commons-lang3'. This is all ready included in master. Sorry for the tangent. I am building it now and will update the PR once I finished my local testing.

**Comment by master-bob** [comment]
Local testing went OK. PR has been updated.

**Pull request merged**